### PR TITLE
feat: create json schema for user-input recipe

### PR DESCRIPTION
**Pull request by nullPointer373**
**Issue #, if available:**

**Description of changes:**
I have added a JSON schema file that defines the user-input recipe format and structure. The schema is based on the existing recipe definition in its Java package and its documentation, which allows for further semantic validation. 

**Why is this change necessary:**
The JSON schema is essential for validating user-input recipes and ensuring they conform to the expected format. By using the schema, we can enforce constraints on the recipe properties and catch any potential errors or invalid input early in the process.

**How was this change tested:**
I have tested the JSON schema by validating it against various recipes manually. And all unit tests (including newly added tests) and integration tests passed successfully.

**Any additional information or context required to review the change:**
One important aspect to consider is the naming convention for recipe properties. While the recipe follows the CamelCase naming convention for properties in its documentation, hands-on experimentation has revealed that the recipe properties can be case-insensitive when specified. To accommodate this behavior, I have deliberately made all property names in the JSON schema lowercase. During the validation process, the property names will be converted to lowercase in the data object before applying the schema. This approach ensures that the validation process is case-insensitive and can handle variations in property casing.

**Checklist:**
- [ ] Updated the README if applicable
- [x] Updated or added new unit tests
- [ ] Updated or added new integration tests
- [ ] Updated or added new end-to-end tests
- [ ] If your code makes a remote network call, it was tested with a proxy
- [ ] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: gdk/static/user_input_recipe_schema.json [hunk 1]
```diff
@@ -0,0 +1,353 @@
+{
+    "$schema": "http://json-schema.org/draft/2020-12/schema",
+    "type": "object",
+    "properties": {
```
Comment (saranyailla) [inline gdk/static/user_input_recipe_schema.json:4]: Are these properties case-insensitive?

File: gdk/static/user_input_recipe_schema.json [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+                "additionalProperties": false
+            }
+        },
+        "lifecycle": {
+            "type": "object"
+        }
+    },
+    "required": ["recipeformatversion", "componentname"],
+    "additionalProperties": false
+}
```
Comment (saranyailla) [inline gdk/static/user_input_recipe_schema.json:353]: can you also add tests for the same?

File: gdk/static/user_input_recipe_schema.json [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+                },
+                "additionalProperties": false
+            }
+        },
+        "lifecycle": {
+            "type": "object"
+        }
+    },
+    "required": ["recipeformatversion", "componentname"],
+    "additionalProperties": false
```
[the remark this answers is not in the document]
Comment (nullPointer373) [reply gdk/static/user_input_recipe_schema.json:352]: Regarding whether we should mark it as true, my suggestion is to set it to false. Enabling this option could lead to errors during the component creation process if there are additional keys beyond the predefined ones. To ensure compatibility with new recipe keys in the future, it's advisable to incorporate them into the schema. This way, the validation process can be extended to accommodate any potential changes.

[the remark this answers is not in the document]
Comment (saranyailla) [reply gdk/static/user_input_recipe_schema.json:352]: Yes, that was my follow up question - if the component creation fails when undefined keys are added. In that case, I agree.

File: gdk/static/user_input_recipe_schema.json (hunk 1 above)
Comment (nullPointer373) [reply gdk/static/user_input_recipe_schema.json:4]: Based on my experience, the case doesn't matter for the recipe properties as long as the property name is accurate.

**Review by saranyailla — DISMISSED** [review]

**Review by saranyailla — APPROVED** [review]

**Review by urvashijain18 — APPROVED** [review]

**Pull request merged**